### PR TITLE
Allow extending INSTALLED_APPS via environment

### PR DESCRIPTION
**Pull request by jovandeginste**
This allows a user to add "apps" (aka parsers) through the environment.

Especially useful when using Docker, and adding a test-parser.

Usage:

```yaml
services:
  webserver:
    environment:
      PAPERLESS_APPS: paperless_tika.apps.PaperlessTikaConfig
```

You can add more by separating them with a `,`:

```yaml
PAPERLESS_APPS: app1,app2
```

### CODE REVIEW

**Comment by jonaswinkler** [comment]
As already stated in the discussion (https://github.com/jonaswinkler/paperless-ng/discussions/199#discussioncomment-247841), this is the wrong approach.

There's no way for users to add source files to docker paperless installations, since the image is immutable. Except for bind mounting folders into the container source directory, which sounds pretty bad.

Rather than that, just add the tika app directly to that list.

**Comment by jovandeginste** [comment]
Yes, you hinted, and I respectfully disagree. Let me make my case:

1. It is an established practice to allow installation of "plugins" through a (or multiple) mounted volume(s); I see a parser as a plugin. I have some examples, not sure it's useful to list them here.
2. It is not only possible through bind mounts; this alternative uses a similar approach:

```Dockerfile
FROM jonaswinkler/paperless-ng:0.9.9
RUN pip install tika
ADD tesseract_tika/ /usr/src/paperless/src/paperless_tika/
```

Now you have a new docker image, with the new parser. To activate the parser, I need to add a complicated, not-so-nice `sed` (didn't actually try this, will probably contain a typo):

```Dockerfile
RUN sed '/^INSTALLED_APPS/a    "paperless_tika.apps.PaperlessTikaConfig",' -i /usr/src/paperless/src/paperless/settings.py
```

So, in conclusion, I do believe this to be a correct and nicer approach. However, it is ultimately your decision, so I rest my case here.

**Comment by jovandeginste** [comment]
(Re: the dev branch, not sure I can re-attach it without losing the discussion here, so I won't touch that for now)

**Comment by jonaswinkler** [comment]
Alright, lets do it this way then.

I've had the following thought in mind:

- Put the parser code in the main repo, and have configuration variables that define the location of the tika and gotenberg server. (such as `PAPERLESS_TIKA_HOST`)
- Have the feature enabled only if `PAPERLESS_TIKA_HOST` is defined.
- This way, docker users would only have to add the additional services and the configuration variables to their docker-compose file to enable the feature. They could also do this with any image they download from the hub, no image building required. Makes the feature MUCH more accessible to other users.
- Users who don't want the feature could simply not configure it and the impact on these users would be minimal (a couple added source files that aren't even getting called).

I'll manually merge this into `dev`, and have this in the next release, which will probably happen around new years.

**Comment by jovandeginste** [comment]
How exactly does one "enable" the feature? I can think of a number of ways, such as returning an empty document in `signals.py`, but there must be a nicer way, no?

**Comment by jonaswinkler** [comment]
I'd simply make

`document_consumer_declaration.connect(tika_consumer_declaration)` in `apps.py`

conditional on some configuration variable. Without this, paperless won't know that your parser exists in the first place.

**Comment by jovandeginste** [comment]
Almost there, just need the place to make `pip install tika` work. I don't seem to find the `requirements.txt`...?

**Comment by jonaswinkler** [comment]
Pipfile.

requirements.txt is generated from that.

**Comment by jovandeginste** [comment]
OK; I'll close this PR and open the new one then.
